### ISSUE TITLE
Should these 2 networks produce a different plot ? (to my eye, they are the same network.)

### ISSUE THREAD

**Issue by ghsanti** [keras-team/keras#19882]
I suspect these 2 models are the same, and the plot just shows the same Input Layer twice, but would like to know if you'd consider these models the same as well ?

```python
from keras.api.layers import Input, Dense
from keras.api.models import Model
from keras.api.utils import plot_model

A1 = Input(shape=(30,),name='A1')
A2 = Dense(8, activation='relu',name='A2')(A1)
A3 = Dense(30, activation='relu',name='A3')(A2)
m1_model = Model(A1, A3)

B1 = Input(shape=(30,),name='B1')
B2 = Dense(40, activation='relu',name='B2')(B1)
B3 = Dense(30, activation='relu',name='B3')(B2)
m2_model = Model(B1, B3)

# method 1: 
m1_out = m1_model(m1_model.inputs[0]) # get list of tensors (with layer H.)
m1m2_out=m2_model(m1_out)
extendedModel_1 = Model(A1,m1m2_out)
extendedModel_1.summary()

# method 2
extendedModel_2 = Model(A1, m2_model(m1_model.outputs))
extendedModel_2.summary()
```

The problem is 
* Plotting `_1`
![image](https://github.com/keras-team/keras/assets/125615505/9dde98a2-df6b-490f-9f6b-b5b71016d1a6)

* Plotting `_2`

![image](https://github.com/keras-team/keras/assets/125615505/c134bdc0-1315-4224-b997-911880501130)

**Comment by mehtamansi29** [comment]
Hi @aka-santi -

The code snippet provided will show only summary of extendedModel_1 and extendedModel_2 model. It will not generate model plots. For generating model plot you need to use [keras.utils.plot_model](https://keras.io/api/utils/model_plotting_utils/) api. 
Using this API, two models(extendedModel_1 and extendedModel_2 model) is generating different plots. Attached [gist](https://colab.sandbox.google.com/gist/mehtamansi29/5984aa19c064a6744d6265e040647255/19882-should-these-2-networks-produce-a-different-plot-to-my-eye-they-are-the-same-network.ipynb) for your reference. 
Let me know if required any other help. Thanks..!

**Comment by aka-santi** [comment]
Thanks, my question is whether those networks are still equivalent, or different

**Comment by mehtamansi29** [comment]
Hi @aka-santi -

Both the model (extendedModel_1 and extendedModel_2) is achieving same functionality with same number of total and trainable parameters. Only different is extendedModel_1 uses (m1_model.input  and m2_model) functional model as intermediate model to create new model while extendedModel_2 uses keras dense layer(output of m1_model)  and m2_model(functional).

**Comment by aka-santi** [comment]
They do have same number of parameters, but the image I added, and the `summary()` with expanded layers is different

**Comment by mehtamansi29** [comment]
Hi @aka-santi-

Can you help me with execution lines which you used for plot the model ?

**Comment by ghsanti** [comment]
Let me rephrase the question, as it can be stated in more simple terms. 

I'm chaining 2 Functional models returned from functions as in `return Model(in,out)`

I understand the idea for chaining them, [as described in the docs.](https://keras.io/guides/functional_api/)

But am trying to learn whether `.outputs` would be equivalent to an instance of the model using `my_model(symbolic_input)`

Here is a minimal idea in code:

```python
in = m1_model.inputs
new_model = Model( in, m2_model( m1_model( in ) ) )
```

or

```python
# .outputs
new_model = Model(m1_model.inputs, m2_model(my_model.outputs))
```

I'm unsure whether `.outputs` is also passing the weights etc and hence it's equivalent to an instance `my_model(symbolic_input)` or not.

**Comment by mehtamansi29** [comment]
Hi @ghsanti -

.outputs doesn't contain weights it contain output tensor only. And both standard way and .outputs satisfying the same result. Only using .outputs, m2_model is processing output of my_model which is the final output of model. While in standard way my_model input is preprocessed by m2_model.
